core/src/main/java/org/apache/brooklyn/core/sensor/ReleaseableLatch.java

```diff
@@ -66,6 +73,13 @@ public MaxConcurrencyLatch(int permits) {
 
             @Override
             public void acquire(Entity caller) {
+                if (!ownerEntities.add(caller)) {
+                    LOG.warn("Entity {} trying to double-acquire permit with ~{} permits available and ~{} threads waiting in queue",
```

Comment:
Perhaps change this log message slightly:
```
LOG.warn("Entity {} acquiring permit multiple times with ~{} permits available ...
```
It could otherwise be mis-understood as saying that the entity was trying to do it, but that it's being skipped.

Worth adding javadoc to say what the semantics are for reentrant calls (and multiple calls by the same entity from different threads).